modular_zubbers/code/modules/security/security_glock/gun.dm

```diff
@@ -1,28 +1,145 @@
+// 	spawn_magazine_type = /obj/item/ammo_box/magazine/m9mm/rubber
+
+/obj/item/gun/ballistic/automatic/pistol/sec_glock/attack_self(mob/living/user)
+	var/obj/item/storage/our_belt = user.get_item_by_slot(ITEM_SLOT_BELT)
+	var/obj/item/ammo_box/magazine/m9mm/security/swapped_mag = locate(/obj/item/ammo_box/magazine/m9mm/security) in our_belt.contents
+	if(!magazine && swapped_mag)
+		if(swapped_mag)
+			if(insert_magazine(user, swapped_mag))
+				SpinAnimation(2, 1)
+				balloon_alert(user, "you spin your gun, loading in a mag!")
+			else
+				balloon_alert(user, "that magazine doesn't fit!")
+		else
+			balloon_alert(user, "no spare magazines in your belt!")
+		return
+	if(!chambered)
+		if(swapped_mag)
+			SpinAnimation(2, 1) //What a badass
+			if(do_after(user, 1 SECONDS, timed_action_flags = ( IGNORE_USER_LOC_CHANGE | IGNORE_TARGET_LOC_CHANGE ), target = src))
+				throw_eject_magazine(user, magejecting)
+		else
+			balloon_alert(user, "no spare magazines in your belt!")
+	if(bolt_type == BOLT_TYPE_NO_BOLT)
+		unload_ammo(user)
+		return
+	drop_bolt_silent(user)
+	if(recent_rack > world.time)
+		return
+	recent_rack = world.time + rack_delay
+	rack_silent(user)
+	return
+
+/obj/item/gun/ballistic/automatic/pistol/sec_glock/proc/throw_eject_magazine(mob/user, display_message = TRUE, obj/item/ammo_box/magazine/tac_load = null, atom/target)
+	if(bolt_type == BOLT_TYPE_OPEN)
+		chambered = null
+	if(magazine.ammo_count())
+		playsound(src, eject_sound, eject_sound_volume, eject_sound_vary)
+	else
+		playsound(src, eject_empty_sound, eject_sound_volume, eject_sound_vary)
+	magazine.forceMove(drop_location())
+	var/obj/item/ammo_box/magazine/old_mag = magazine
+	var/obj/item/storage/our_belt = user.get_item_by_slot(ITEM_SLOT_BELT)
+	var/obj/item/ammo_box/magazine/m9mm/security/swapped_mag = locate(/obj/item/ammo_box/magazine/m9mm/security) in our_belt.contents
+	if(swapped_mag)
+		insert_magazine(user, swapped_mag)
+	else
+		magazine = null
+		return
+	old_mag.forceMove(drop_location())
+	if(istype(old_mag, /obj/item/ammo_box/magazine/m9mm/security/rocket) && magejecting)
+		playsound(old_mag, 'sound/items/weapons/gun/general/rocket_launch.ogg', 50, TRUE)
+	if(magejecting)
+		old_mag.throw_at(get_edge_target_turf(user, user.dir), range = 7, speed = 7, thrower = user, force = 10)
+	old_mag.update_appearance()
+	update_appearance()
 
-/obj/item/gun/ballistic/automatic/pistol/sec_glock/security/rubber //This is what you give to cargo packages.
-	spawn_magazine_type = /obj/item/ammo_box/magazine/m9mm/rubber
+/obj/item/gun/ballistic/automatic/pistol/sec_glock/insert_magazine(mob/user, obj/item/ammo_box/magazine/AM, display_message = TRUE)
+	if(!istype(AM, accepted_magazine_type))
+		balloon_alert(user, "[AM.name] doesn't fit!")
+		return FALSE
+	user.transferItemToLoc(AM, src)
+	magazine = AM
+	if(bolt_type == BOLT_TYPE_OPEN && !bolt_locked)
+		chamber_round(TRUE)
+	drop_bolt_silent()
+	update_appearance()
+	.=..()
```

Comment:
missing space after .
Plus, you're setting the parent return value to the default return value already, yet overriding it on the next line with TRUE, consider checking parent sanity somehow before reloading